models/user/email_address.go

```diff
@@ -142,18 +141,12 @@ func (email *EmailAddress) BeforeInsert() {
 	}
 }
 
-var emailRegexp = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]*@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
-
 // ValidateEmail check if email is a allowed address
 func ValidateEmail(email string) error {
 	if len(email) == 0 {
 		return nil
 	}
 
-	if !emailRegexp.MatchString(email) {
-		return ErrEmailCharIsNotSupported{email}
-	}
-
```

Comment:
This just disables all checks. Is this really what is wanted? A more complete regex is:

(?:[a-z0-9!#$%&'+/=?^_{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_{|}~-]+)|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\[\x01-\x09\x0b\x0c\x0e-\x7f])")@(?:(?:a-z0-9?.)+a-z0-9?|[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])).){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-][a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\[\x01-\x09\x0b\x0c\x0e-\x7f])+)])

https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression/14075810#14075810

Reply:
no - regex are not a legitimate means of validating email addresses

sending a verification email is the means to validate email addresses.

https://github.com/go-gitea/gitea/issues/19852#issuecomment-1142533132

Reply:
why not use the net/mail library and mail.ParseAddress(...) to test if it conforms to RFC5322

Reply:
> why not use the net/mail library and mail.ParseAddress(...) to test if it conforms to RFC5322

@eeyrjmr , are you asking why this wasn't deleted: https://github.com/go-gitea/gitea/blob/433d6845db2f7634fe4f1f37c000af0df54d0155/models/user/email_address.go#L154

or was this a hypothetical question?